### PR TITLE
Fix qr address import

### PR DESCRIPTION
**Pull request by joelamouche**
solves https://github.com/polkadot-js/apps/issues/5589#issuecomment-862105778

Modify add qr code to 
- reject custom metadata and 
- accept ethereum addresses (depends on https://github.com/polkadot-js/ui/pull/484/ )

### CODE REVIEW

File: packages/page-accounts/src/modals/Qr.tsx [hunk 1]
```diff
@@ -138,8 +143,12 @@ function QrModal ({ className = '', onClose, onStatusChange }: Props): React.Rea
           : (
             <Modal.Columns hint={t<string>('Provide the account QR from the module/external application for scanning. Once detected as valid, you will be taken to the next step to add the account to your list.')}>
               <div className='qr-wrapper'>
-                <QrScanAddress onScan={_onScan} />
+                <QrScanAddress isEthereum={isEthereum}
+                  onScan={_onScan} />
               </div>
+              {
+                warning && <MarkWarning>{warning}</MarkWarning>
+              }
```
Comment (jacogr) [inline packages/page-accounts/src/modals/Qr.tsx:151]:
```suggestion
              {warning && <MarkWarning content={warning} />}
```

The component is a bit "funny" in this regard. (Still not sure if it is a good idea or a bad one atm...)

File: packages/page-accounts/src/modals/Qr.tsx [hunk 2]
```diff
@@ -138,8 +143,12 @@ function QrModal ({ className = '', onClose, onStatusChange }: Props): React.Rea
           : (
             <Modal.Columns hint={t<string>('Provide the account QR from the module/external application for scanning. Once detected as valid, you will be taken to the next step to add the account to your list.')}>
               <div className='qr-wrapper'>
-                <QrScanAddress onScan={_onScan} />
+                <QrScanAddress isEthereum={isEthereum}
+                  onScan={_onScan} />
```
Comment (jacogr) [inline packages/page-accounts/src/modals/Qr.tsx:147]:
```suggestion
                <QrScanAddress
                  isEthereum={isEthereum}
                  onScan={_onScan}
                />
```

File: packages/page-accounts/src/modals/Qr.tsx [hunk 3]
```diff
@@ -55,19 +56,23 @@ function QrModal ({ className = '', onClose, onStatusChange }: Props): React.Rea
...
     (scanned: Scanned): void => {
-      setAddress({
-        address: scanned.isAddress
-          ? scanned.content
-          : keyring.createFromUri(scanned.content, {}, 'sr25519').address,
-        isAddress: scanned.isAddress,
-        scanned
-      });
+      if (isEthereum && scanned.isAddress && scanned.content.substring(0, 2) !== '0x') {
+        setAddress({ address: '', isAddress: false, scanned: null, warning: 'Ethereum Compatible Parachains must import from an Ethereum Wallet' });
```
Comment (jacogr) [inline packages/page-accounts/src/modals/Qr.tsx:60]:
```suggestion
        setAddress({
          address: '',
          isAddress: false,
          scanned: null,
          warning: t<string>('Ethereum Compatible Parachains must import from an Ethereum Wallet')
        });
```

Translatable strings

File: packages/page-accounts/src/modals/Qr.tsx [hunk 4]
```diff
@@ -55,19 +56,23 @@ function QrModal ({ className = '', onClose, onStatusChange }: Props): React.Rea
...
+          scanned
+        });
+      }
 
       if (scanned.name) {
         _onNameChange(scanned.name);
       }
     },
-    [_onNameChange]
+    [_onNameChange, isEthereum]
```
Comment (jacogr) [inline packages/page-accounts/src/modals/Qr.tsx:75]:
```suggestion
    [_onNameChange, isEthereum, t]
```

Since we use `t` above for the warning string

**Review by jacogr — COMMENTED** [review]
Some small comments.

**Comment by joelamouche** [comment]
@jacogr lmk when the @polkadot/react-qr is updated

**Comment by jacogr** [comment]
It is updated, the PR was linked through.

**Comment by joelamouche** [comment]
https://www.npmjs.com/package/@polkadot/react-qr version didnt change

**Comment by jacogr** [comment]
It did https://www.npmjs.com/package/@polkadot/react-qr 0.8.2-2 as per the linked PR.

**Comment by joelamouche** [comment]
oh okay thanks

**Comment by joelamouche** [comment]
Necessary PRs: polkadot-js/common#1032 and polkadot-js/ui#491

**Comment by joelamouche** [comment]
@jacogr ready for review

**Comment by jacogr** [comment]
I think the build fails due to the libs probably not on latest (last UI PR is not in yet, I am going through updating now... although, not sure it would fail, but anyway, we need latest as it stans)

**Comment by jacogr** [comment]
It has a couple of build failures where it seems an assumption is made to the type, which is those cases are not applicable. (TS finds them as `never`, and they are treated as strings). Have not looked into the causes at depth.

**Comment by joelamouche** [comment]
all green @jacogr

File: packages/page-accounts/src/modals/Qr.tsx [hunk 5]
```diff
@@ -70,6 +71,18 @@ function QrModal ({ className = '', onClose, onStatusChange }: Props): React.Rea
     [_onNameChange]
   );
 
+  const _onError = useCallback(
+    (err: Error): void => {
+      setAddress({
+        address: '',
+        isAddress: false,
+        scanned: null,
+        warning: t<string>(err.message)
```
Comment (jacogr) [inline packages/page-accounts/src/modals/Qr.tsx:80]: Small comment, since this is an error (non-translatable), should just be `err.message`

**Review by jacogr — APPROVED** [review]
Seems great, just one small adjustment to merge

**Comment by joelamouche** [comment]
updated

**Review by jacogr — APPROVED** [review]
Thank you.

**Pull request merged**